### PR TITLE
perf: question validate

### PR DESCRIPTION
**Pull request by wenytang-ms**

### CODE REVIEW

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -260,6 +260,7 @@
   "plugins.apiConnector.APIKeyOption.detail": "Take API Key as the way to authentication",
   "plugins.apiConnector.ImplementMyselfOption.detail": "Define the authentication by yourself",
   "plugins.apiConnector.questionComponentSelect.emptySelection": "You have to select at least one component.",
+  "plugins.apiConnector.QuestionAppName.validation.ApiNameExist": "Exists a file with the same name, please rename it",
```
Comment (xzf0587) [inline packages/fx-core/resource/package.nls.json:263]: specify the exact conflicted file name to help user understand.

File: packages/fx-core/src/plugins/resource/apiconnector/questions.ts [hunk 2]
```diff
@@ -1,14 +1,36 @@
...
+      const apiNames: string[] = previousInputs![
+        Constants.questionKey.componentsSelect
+      ] as string[];
+      const projectPath: string = previousInputs?.projectPath as string;
+      for (const apiName of apiNames) {
+        const componentPath = path.join(projectPath, apiName);
+        const JsFileName = input + "." + FileType.JS;
+        const TsFileName = input + "." + FileType.TS;
+        if (
+          (await fs.pathExists(path.join(componentPath, JsFileName))) ||
```
Comment (xzf0587) [inline packages/fx-core/src/plugins/resource/apiconnector/questions.ts:25]: here we check the js and ts file both. should we check the necessary one?

File: packages/fx-core/src/plugins/resource/apiconnector/questions.ts [hunk 3]
```diff
@@ -1,14 +1,36 @@
...
+import { Constants, FileType } from "./constants";
 import { getLocalizedString } from "../../../common/localizeUtils";
 
 export const apiNameQuestion: TextInputQuestion = {
   name: Constants.questionKey.apiName,
   title: getLocalizedString("plugins.apiConnector.getQuestionApiName.title"),
   type: "text",
+  validation: {
+    validFunc: async (input: string, previousInputs?: Inputs): Promise<string | undefined> => {
+      const apiNames: string[] = previousInputs![
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/resource/apiconnector/questions.ts:16]: better variable naming? maybe `components` or something else

File: packages/fx-core/resource/package.nls.json [hunk 4]
```diff
@@ -260,7 +260,7 @@
   "plugins.apiConnector.APIKeyOption.detail": "Take API Key as the way to authentication",
   "plugins.apiConnector.ImplementMyselfOption.detail": "Define the authentication by yourself",
   "plugins.apiConnector.questionComponentSelect.emptySelection": "You have to select at least one component.",
-  "plugins.apiConnector.QuestionAppName.validation.ApiNameExist": "Exists a file with the same name, please rename it",
+  "plugins.apiConnector.QuestionAppName.validation.ApiNameExist": "Please rename a different API name to avoid conflicts with existing file %s",
```
Comment (blackchoey) [inline packages/fx-core/resource/package.nls.json:263]: Please `provide` a different API name ... ?

**Review by xzf0587 — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/apiconnector/questions.ts [hunk 5]
```diff
@@ -3,35 +3,75 @@
...
+      name: Constants.questionKey.apiName,
+      title: getLocalizedString("plugins.apiConnector.getQuestionApiName.title"),
+      validation: {
+        validFunc: async (input: string, previousInputs?: Inputs): Promise<string | undefined> => {
+          const fileEx =
+            this.ctx?.projectSetting.programmingLanguage === LanguageType.JS
+              ? FileType.JS
+              : FileType.TS;
+          const projectPath: string = previousInputs?.projectPath as string;
+          const apiFileName: string = input + "." + fileEx;
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/resource/apiconnector/questions.ts:56]: find some way to share the file name generation logic with other part?

File: packages/fx-core/resource/package.nls.json (hunk 4 above)
Comment (wenytang-ms) [reply packages/fx-core/resource/package.nls.json]: sure

File: packages/fx-core/src/plugins/resource/apiconnector/questions.ts (hunk 5 above)
Comment (wenytang-ms) [reply packages/fx-core/src/plugins/resource/apiconnector/questions.ts]: updated

**Review by blackchoey — APPROVED** [review]

**Pull request merged**